src/libstd/net/udp.rs

```diff
@@ -103,25 +105,35 @@ impl UdpSocket {
         super::each_addr(addr, net_imp::UdpSocket::bind).map(UdpSocket)
     }
 
-    /// Receives data from the socket. On success, returns the number of bytes
-    /// read and the address from whence the data came.
+    /// Receives single datagram message from the socket. On success, returns the number
+    /// of bytes read and the address from whence the data came.
+    /// The function must be called with valid byte array `buf` of sufficient size to
+    /// hold the message bytes. If a message is too long to fit in the supplied buffer,
+    /// excess bytes may be discarded.
```

Comment:
[Quoting from the documentation documentation:](https://doc.rust-lang.org/book/first-edition/documentation.html#writing-documentation-comments)

> The first line of a documentation comment should be a short summary of its functionality. One sentence. Just the basics. High level.

(I think by "line" they mean "paragraph" -- everything before the first blank line. The first paragraph is special in rustdoc. For example it is visible in the module-level documentation before you click through on the function.)

Everything you wrote is important to document but not all of it is high level basics. Could you split this into two paragraphs? The first one should give a general understanding of what the function does, and the second fills in the details.